java/arcs/core/util/Guard.kt

```diff
@@ -70,15 +72,20 @@ class Guard<T>(private val mutex: Mutex, private val initialValue: T) {
  */
 private class GuardDelegate<T>(
     private val mutex: Mutex,
-    private var value: T
+    private val initialValue: () -> T
 ) : ReadWriteProperty<Any, T> {
+    private var valueHolder: ValueHolder<T>? = null
```

Comment:
Is the goal of this boxing to make it possible to set `null` on optional types while still preventing the initializer from running? (A test to verify setting `null` prevents initializer run might be nice to add too, in that case?)

Reply:
That's right

Reply:
And great test idea - that caught a bug! Thanks!